model-engine/model_engine_server/inference/forwarding/http_forwarder.py

```diff
@@ -181,23 +182,30 @@ def add_extra_routes(app: FastAPI):
         all_routes = set(list(sync_forwarders.keys()) + list(stream_forwarders.keys()))
 
         for route in all_routes:
+
+            def get_sync_forwarder(route=route):
+                return sync_forwarders.get(route)
+
+            def get_stream_forwarder(route=route):
+                return stream_forwarders.get(route)
+
             # This route is a catch-all for any requests that don't match the /predict or /stream routes
             # It will treat the request as a streaming request if the "stream" body parameter is set to true
             # NOTE: it is important for this to be defined AFTER the /predict and /stream endpoints
             # because FastAPI will match the first route that matches the request path
             async def predict_or_stream(
                 request: EndpointPredictV1Request,
                 background_tasks: BackgroundTasks,
-                sync_forwarder=Depends(lambda: sync_forwarders.get(route)),
-                stream_forwarder=Depends(lambda: stream_forwarders.get(route)),
+                sync_forwarder=Depends(get_sync_forwarder),
+                stream_forwarder=Depends(get_stream_forwarder),
                 limiter=Depends(get_concurrency_limiter),
             ):
                 if not request.args:
                     raise Exception("Request has no args")
                 if request.args.root.get("stream", False) and stream_forwarder:
                     return await stream(request, stream_forwarder, limiter)
                 elif request.args.root.get("stream") is not True and sync_forwarder:
-                    return predict(request, background_tasks, sync_forwarder, limiter)
+                    return await predict(request, background_tasks, sync_forwarder, limiter)
```

Comment:
this was a big woopsie; this is a blocking method in an async function which essentially blocks the thread and severely restricting concurrency